packages/gestalt/src/MenuOption.js

```diff
@@ -0,0 +1,229 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Badge from './Badge.js';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Link from './Link.js';
+import Text from './Text.js';
+import styles from './Touchable.css';
+import getRoundingClassName from './getRoundingClassName.js';
+import Icon from './Icon.js';
+import focusStyles from './Focus.css';
+import useFocusVisible from './useFocusVisible.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+  subtext?: string,
+|};
+
+type OptionProps = {|
+  badgeText?: string,
+  children?: Node,
+  index: number,
+  option: OptionObject,
+  selected?: OptionObject | Array<OptionObject> | null,
+  handleSelect?: ({|
+    item: OptionObject,
+    event: SyntheticFocusEvent<HTMLInputElement>,
+  |}) => void,
+  hoveredItem: ?number,
+  isExternal?: boolean,
+  role?: 'option' | 'menuitem',
+  setHoveredItem: (number) => void,
+  setOptionRef: (?HTMLElement) => void,
+  shouldTruncate?: boolean,
+  textWeight?: 'bold' | 'normal',
+  url?: string,
+|};
+
+export default function MenuOption({
+  badgeText,
+  children,
+  handleSelect,
+  hoveredItem,
+  index,
+  isExternal,
+  option,
+  role,
+  selected,
+  setHoveredItem,
+  setOptionRef,
+  shouldTruncate = false,
+  textWeight = 'normal',
+  url,
+}: OptionProps): Node {
+  let foundAMatch = [];
+  if (selected instanceof Array)
+    foundAMatch = selected.filter((item) => item.value === option.value);
+
+  // Determine if the option is a current selected item
+  const isSelectedItem =
+    foundAMatch.length > 0 ||
+    JSON.stringify(option) === JSON.stringify(selected);
+
+  const handleOnTap = (event) => {
+    if (!url && !children) {
+      event.preventDefault();
+    }
+    if (handleSelect) handleSelect({ event, item: option });
+  };
+
+  const { isFocusVisible } = useFocusVisible();
+
+  const className = classnames(
+    getRoundingClassName(2),
+    focusStyles.hideOutline,
+    {
+      [focusStyles.accessibilityOutline]: isFocusVisible,
+      [styles.fullWidth]: true,
+      [styles.pointer]: true,
+    }
+  );
+
+  // Default option color
+  let optionStateColor = 'transparent';
+
+  // Set color on item hover
+  if (index === hoveredItem) optionStateColor = 'lightGray';
+
+  const menuOptionContents = (
+    <Flex>
+      <Flex flex="grow" direction="column">
+        <Flex alignItems="center">
+          {children || (
+            <>
+              <Text
+                truncate={shouldTruncate}
+                weight={textWeight}
+                color="darkGray"
+                inline
+              >
+                {option?.label}
+              </Text>
+              {badgeText && (
+                <Box marginStart={2} marginTop={1}>
+                  <Box display="visuallyHidden">{`, `}</Box>
```

Comment:
I assume this helps with accessibility? What does this do? (It might be worth commenting in the code for documentation)

Reply:
It prevents a run on sentence b/w the item label and the badge text (will add a comment)

Reply:
Oh interesting! That seems like a good trick to document in an a11y best practices doc somewhere